noxfile.py

```diff
@@ -0,0 +1,212 @@
+"""
+Configuration for setting up virtual environments to run tests, build
+documentation, and check code style.
+
+Use 'nox -l' to see a list of all sessions available and 'nox -s SESSION_NAME'
+to run a session. Use '-rs' instead of '-s' to avoid creating new virtual
+environments all the time.
+"""
+from pathlib import Path
+import shutil
+
+import nox
+
+
+PACKAGE = "boule"
+PYTHON_VERSIONS = ["3.8", "3.7", "3.6"]
+DOCS = Path("doc")
+REQUIREMENTS = {
+    "run": "requirements.txt",
+    "test": str(Path("env") / "requirements-test.txt"),
+    "docs": str(Path("env") / "requirements-docs.txt"),
+}
+
+PYLINT_FILES = [PACKAGE, "setup.py"]
+FLAKE8_FILES = [PACKAGE, "setup.py", str(DOCS / "conf.py")]
+BLACK_FILES = [PACKAGE, "setup.py", str(DOCS / "conf.py"), "tutorials", "noxfile.py"]
+
+# Use absolute paths for pytest otherwise it struggles to pick up coverage info
+PYTEST_ARGS = [
+    f"--cov-config={Path('.coveragerc').resolve()}",
+    "--cov-report=term-missing",
+    f"--cov={PACKAGE}",
+    "--doctest-modules",
+    '--doctest-glob="*.rst"',
+    "-v",
+    "--pyargs",
+]
+RST_FILES = [str(p.resolve()) for p in Path("doc").glob("**/*.rst")]
+
+
+# Configure Nox
+nox.options.sessions = ("black_check", "flake8", "pylint", "test")
+
+
+@nox.session()
+def black_check(session):
+    """
+    Check code style using black
+    """
+    session.install("black")
+    list_packages(session)
+    session.run("black", "--check", *BLACK_FILES)
+
+
+@nox.session()
+def flake8(session):
+    """
+    Check code style using flake8
+    """
+    session.install("flake8")
+    list_packages(session)
+    session.run("flake8", *FLAKE8_FILES)
+
+
+@nox.session()
+def pylint(session):
+    """
+    Check code style using pylint
+    """
+    session.install("pylint==2.4.*")
+    list_packages(session)
+    session.run("pylint", *PYLINT_FILES)
+
+
+@nox.session()
+def format(session):
+    """
+    Run black to format the code
+    """
+    session.install("black")
+    session.run("black", *BLACK_FILES)
+
+
+def build_and_install(session):
+    """
+    Build source and wheel packages for the project and install them
+    """
+    if Path("dist").exists():
+        shutil.rmtree("dist")
+    session.install("wheel")
+    session.run("python", "setup.py", "sdist", "bdist_wheel", silent=True)
+    wheel = [str(p) for p in Path("dist").glob("*.whl")]
+    assert len(wheel) == 1, f"More than 1 wheel present: {wheel}"
+    session.install("--no-deps", wheel[0])
```

Comment:
I'm getting an error from this assertion when running `nox --install-only` because the `wheel` is an empty list.

```suggestion
    if wheel:
        assert len(wheel) == 1, f"More than 1 wheel present: {wheel}"
        session.install("--no-deps", wheel[0])
```